src/v2/modules/onboarding/index.ts

```diff
@@ -81,17 +80,17 @@ async function playCatchup(guild) {
   }
 }
 
-function addNewRolePermissions(guild, role) {
-  const perms = new Permissions(Permissions.DEFAULT);
-  perms.remove(Permissions.FLAGS.VIEW_CHANNEL);
+function addNewRolePermissions(guild: Guild, role: Role) {
+  const permissionFlags = PermissionsBitField.Default & ~PermissionsBitField.Flags.ViewChannel
```

Comment:
question: What's with the `~PermissionsBitField.Flags.ViewChannel` particually the `~`

Reply:
Discord permissions work with a Bit Fields (which are bigints that store permissions encoded as individual bits) and discord.js exposes this now, so doing boolean maths here. 

I want to use the default permissions but remove the ability to view the channel.

`~` is a [bitwise `NOT`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Bitwise_NOT), it flips every bit.
`&` is a [bitwise `AND`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Bitwise_AND), it ANDs every bit between two numbers

Lets say the default permissions are `0b0111`and the view channel permission is `0b0010`:
1. I `NOT` the view channel perms (turning `0b0010` into `0b1101`)
2. I `AND` the result with the default perms so `0b0111 & 01101` which results in `0b0101`

---

I'm not sure, although I wouldn't be surprised if there are some helper functions, particularly under `PermissionBitField` that could allow me to refactor this. Considering that bitwise stuff isn't as common in the mainstream JS world, do you think I should refactor this to potentially use that? Or should I add a comment with the explanation?

Reply:
Ah nice thanks for explaining it to me! should defo spruce up my discord bot knowledge been a few years